File: apps/patter-voice-sdk/libraries/python/getpatter/tts/rime.py
"""Rime TTS for Patter pipeline mode."""

from __future__ import annotations

import os
from typing import ClassVar, Optional

from getpatter.providers.rime_tts import RimeTTS as _RimeTTS

__all__ = ["TTS"]


class TTS(_RimeTTS):
    """Rime HTTP TTS (Arcana / Mist models).

    Example::

        from getpatter.tts import rime

        tts = rime.TTS()                    # reads RIME_API_KEY
        tts = rime.TTS(api_key="...", speaker="astra")
    """

    provider_key: ClassVar[str] = "rime"

    def __init__(
        self,
        api_key: str | None = None,
        *,
        model: str = "arcana",
        speaker: Optional[str] = None,
        lang: str = "eng",
        sample_rate: int = 16000,
    ) -> None:
        key = api_key or os.environ.get("RIME_API_KEY")
        if not key:
            raise ValueError(
                "Rime TTS requires an api_key. Pass api_key='...' or "
                "set RIME_API_KEY in the environment."
            )
        super().__init__(
            api_key=key,
            model=model,
            speaker=speaker,
            lang=lang,
            sample_rate=sample_rate,
        )

File: apps/patter-voice-sdk/libraries/typescript/src/tts/rime.ts
/** Rime TTS for Patter pipeline mode. */
import { RimeTTS as _RimeTTS } from "../providers/rime-tts";

/** Constructor options for the Rime `TTS` adapter. */
export interface RimeTTSOptions {
  /** API key. Falls back to RIME_API_KEY env var when omitted. */
  apiKey?: string;
  model?: string;
  speaker?: string;
  lang?: string;
  sampleRate?: number;
  repetitionPenalty?: number;
  temperature?: number;
  topP?: number;
  maxTokens?: number;
  speedAlpha?: number;
  reduceLatency?: boolean;
  pauseBetweenBrackets?: boolean;
  phonemizeBetweenBrackets?: boolean;
  baseUrl?: string;
}

/**
 * Rime TTS (Arcana or Mist models).
 *
 * @example
 * ```ts
 * import * as rime from "getpatter/tts/rime";
 * const tts = new rime.TTS();              // reads RIME_API_KEY
 * const tts = new rime.TTS({ apiKey: "...", speaker: "astra" });
 * ```
 */
export class TTS extends _RimeTTS {
  static readonly providerKey = "rime";
  constructor(opts: RimeTTSOptions = {}) {
    const key = opts.apiKey ?? process.env.RIME_API_KEY;
    if (!key) {
      throw new Error(
        "Rime TTS requires an apiKey. Pass { apiKey: '...' } or " +
          "set RIME_API_KEY in the environment.",
      );
    }
    const { apiKey: _ignored, ...rest } = opts;
    void _ignored;
    super(key, rest);
  }
}
